src/Command/Analyze.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace HalloWelt\MigrateConfluence\Command;
+
+use Exception;
+use HalloWelt\MediaWiki\Lib\Migration\Command\Analyze as CommandAnalyze;
+use HalloWelt\MediaWiki\Lib\Migration\IAnalyzer;
+use HalloWelt\MediaWiki\Lib\Migration\IOutputAwareInterface;
+use HalloWelt\MediaWiki\Lib\Migration\IUserInteraction;
+use Symfony\Component\Console\Input\InputDefinition;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Yaml\Exception\ParseException;
+use Symfony\Component\Yaml\Yaml;
+
+class Analyze extends CommandAnalyze {
+
+	/**
+	 *
+	 * @inheritDoc
+	 */
+	protected function configure() {
+		$config = parent::configure();
+
+		/** @var InputDefinition */
+		$definition = $this->getDefinition();
+		$definition->addOption(
+			new InputOption(
+				'config',
+				null,
+				InputOption::VALUE_REQUIRED,
+				'Specifies the path to the config yaml file'
+			)
+		);
+
+		return $config;
+	}
+
+	/**
+	 * @param array $config
+	 * @return Analyze
+	 */
+	public static function factory( $config ): Analyze {
+		return new static( $config );
+	}
+
+	/**
+	 * @return bool
+	 */
+	protected function doProcessFile(): bool {
+		$analyzerFactoryCallbacks = $this->config['analyzers'];
+		$this->readConfigFile( $this->config );
+		foreach ( $analyzerFactoryCallbacks as $key => $callback ) {
+			$analyzer = call_user_func_array(
+				$callback,
+				[ $this->config, $this->workspace, $this->buckets ]
+			);
+			if ( $analyzer instanceof IAnalyzer === false ) {
+				throw new Exception(
+					"Factory callback for analyzer '$key' did not return an "
+					. "IAnalyzer object"
+				);
+			}
+			if ( $analyzer instanceof IOutputAwareInterface ) {
+				$analyzer->setOutput( $this->output );
+			}
+			if ( $analyzer instanceof IUserInteraction ) {
+				$analyzer->setQuestionHelper( $this->getHelper( 'question' ) );
+				$analyzer->setOutput( $this->output );
+				$analyzer->setInput( $this->input );
+			}
```

Comment:
I still believe this should be done by the base class. This subclass should just do something like 

```php
$this->readConfigFile( $this->config );
return parent::doProcessFile();
```